.gitlab-ci.yml

```diff
@@ -23,17 +26,19 @@ cache:
 
 before_script:
   - python -V  # Print out python version for debugging
-  - pip install -r requirements.txt
-  - pip install python-coveralls
+  - pip install -r requirements.txt python-coveralls pylint pycodestyle py.test
 
 test:
   stage: test
   script:
-    - echo $TEST_SUITE
-    - echo $TEST_SUITE > test.sh
-    - bash test.sh
-    - coverage report
-  #  - coveralls
+    # TODO bjuhasz: use tox because "setup.py test" is deprecated:
+    # https://setuptools.readthedocs.io/en/latest/setuptools.html#test-build-package-and-run-a-unittest-suite
+    - coverage run --branch --source=pyega3 -m py.test
+    - coverage report --fail-under=$COVERAGE_FAIL_UNDER
+    - pylint --ignore=test/support --fail-under=$PYLINT_FAIL_UNDER pyega3/ test/
```

Comment:
@afoix, please run this command on your local machine and consider fixing those issues which you find important. Or if that was too much, then maybe we should create a follow-up ticket.